Gruntfile.js

```diff
@@ -40,6 +40,24 @@ var getFromExec = function (command, options) {
     return result;
 };
 
+/**
+ * Rename function for grunt file tasks for  adding ".min" convention in front
+ * of minified files; won't do anything to files that already have ".min" in
+ * their filename
+ * @param {String} dest - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+ * @param {String} src - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+ * @param {String} extension - the extension of the file in question ("js" or "css")
+*/
+var addMinifyToFilename = function (dest, src, extension) {
+    var minifiedExtension = ".min." + extension;
+    // Don't operate on files that already have a .min convention
+    if (src.indexOf(minifiedExtension) > -1) {
+        return dest + src;
+    } else {
+        return dest + src.replace("." + extension, minifiedExtension);
+    }
+};
```

Comment:
I wonder if it would be safer and easier to use node's [path](https://nodejs.org/dist/latest-v6.x/docs/api/path.html) module? I don't think you'd need to take an extension as an argument in that case, as you could source it from the filename itself. You could also avoid the replace call at the end but just reconstructing the file name with it's name, "min" and extension.